engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsNode.java

```diff
@@ -133,17 +146,51 @@ boolean equalsDoubleBigInt(double self, EnsoBigInteger other) {
 
   @Specialization
   boolean equalsUnresolvedSymbols(UnresolvedSymbol self, UnresolvedSymbol otherSymbol,
-                                  @Cached EqualsAnyNode equalsNode) {
+                                  @Cached EqualsNode equalsNode) {
     return self.getName().equals(otherSymbol.getName())
         && equalsNode.execute(self.getScope(), otherSymbol.getScope());
   }
 
   @Specialization
   boolean equalsUnresolvedConversion(UnresolvedConversion selfConversion, UnresolvedConversion otherConversion,
-                                     @Cached EqualsAnyNode equalsNode) {
+                                     @Cached EqualsNode equalsNode) {
     return equalsNode.execute(selfConversion.getScope(), otherConversion.getScope());
   }
 
+  @Specialization
+  boolean equalsModuleScopes(ModuleScope selfModuleScope, ModuleScope otherModuleScope,
+      @Cached EqualsNode equalsNode) {
+    return equalsNode.execute(selfModuleScope.getModule(), otherModuleScope.getModule());
+  }
+
+  @Specialization
+  @TruffleBoundary
+  boolean equalsModules(Module selfModule, Module otherModule,
+      @Cached EqualsNode equalsNode) {
+    return equalsNode.execute(selfModule.getName().toString(), otherModule.getName().toString());
```

Comment:
Very likely two modules are equal when `selfModule == otherModule`.

Reply:
As mentioned in other comments, this is for consistency with `HashCodeNode`, where for a `Module` is computed from its name.